.changeset/silver-rivers-judge.md

```diff
@@ -0,0 +1,39 @@
+---
+"fluid-framework": minor
+"@fluidframework/tree": minor
+---
+
+Enforce use of TreeViewConfiguration's constructor
+
+`TreeViewConfiguration` is `@sealed` meaning creating custom implementations of it such as assigning object literals to a `TreeViewConfiguration` or sub-classing it are not supported.
+This reserved the ability for the authors of it (Fluid Framework) to add members.
+This policy however was only indicated via documentation,
+and an implicit assumption that when an API takes in a typed defined as a class, that an instance of that class must be used.
```

Comment:
```suggestion
`TreeViewConfiguration` is `@sealed`, meaning creating custom implementations of it such as assigning object literals to a `TreeViewConfiguration` or sub-classing it are not supported.
This preserves the ability for the Fluid Framework to add members to this class over time without breaking downstream code.
However, there was no code-level enforcement of this expectation. It was only indicated via documentation
and an implicit assumption that when an API takes in a typed defined as a class, that an instance of that class must be used rather than an arbitrary object of a similar shape.
```

Reply:
I specifically used the past tense here to talk about things that were true before this PR to avoid confusion around what this PR is doing. It would be a breaking change for this PR to add this policy, but it is (probably) ok for it to enforce/rely on it and I want to make it clear which parts of this are describing existing policy, and which parts are changes.



```suggestion
`TreeViewConfiguration` is `@sealed`, meaning creating custom implementations of it such as assigning object literals to a `TreeViewConfiguration` or sub-classing it are not supported.
This reserved the ability for the Fluid Framework to add members to this class over time, informing users that they must use it in such a way where such changes are non-breaking.
However, there was no compiler-based enforcement of this expectation.
It was only indicated via documentation and an implicit assumption that when an API takes in a typed defined as a class, that an instance of that class must be used rather than an arbitrary object of a similar shape.
```